### DISCUSSION TITLE
ModifiedByCalendar doesn't work

### DISCUSSION THREAD

**Discussion by ybarbaria** [quartznet/quartznet#489] [General] [answered]
Hello, 

I try to create a trigger with excluded date and included date with quartz but when I get all dates generated by the trigger, my excluded date are inside the trigger.

Someone knows why?

My sample code :

        ScheduleListener schedListen = new ScheduleListener();
        HolidayCalendar calFr = new HolidayCalendar();
        var publicHolidaysFr = DateSystem.GetPublicHoliday("FR", 2017);
        foreach (var publicHoliday in publicHolidaysFr)
        {
            calFr.AddExcludedDate(publicHoliday.Date.ToUniversalTime());
        }

        HolidayCalendar calUS = new HolidayCalendar();
        var publicHolidaysUS = DateSystem.GetPublicHoliday("US", 2017);
        foreach (var publicHoliday in publicHolidaysUS)
        {
            calUS.AddExcludedDate(publicHoliday.Date.ToUniversalTime());
        }

        //Adding weekly calendar to trigger on Monday and Tuesday.
        var weeklyCalendar = new WeeklyCalendar();
        weeklyCalendar.SetDayExcluded(DayOfWeek.Monday, false);
        weeklyCalendar.SetDayExcluded(DayOfWeek.Tuesday, false);

        ISchedulerFactory schedulerFactory = new StdSchedulerFactory();
        IScheduler scheduler = schedulerFactory.GetScheduler();
        scheduler.AddCalendar("calendarPublicHolidaysFr", calFr, true, true);
        scheduler.AddCalendar("calendarPublicHolidaysUK", calUS, true, true);
        scheduler.Start();

        ITrigger trigger1 = TriggerBuilder.Create()
            .WithIdentity("myTriggerHolidays")
            .WithSchedule(CronScheduleBuilder.DailyAtHourAndMinute(16, 08))
            .ModifiedByCalendar("calendarPublicHolidaysFr")
            .ModifiedByCalendar("calendarPublicHolidaysUK") // Mais pas les jours fériés, exclus dans le calendrier précédent.
            .Build();

        var dates = TriggerUtils.ComputeFireTimes(trigger1 as IOperableTrigger, null, 1000);
        foreach (var dateTimeOffset in dates)
        {
            Console.WriteLine(dateTimeOffset.ToString("yyyy-M-d dddd"));
        }
`

**Comment by lahma** [comment, marked as answer]
Triggers support only single calendar, so you need to chain your calendars if you need to have multiple calendars to take effect on a trigger.
